src/main/java/io/jenkins/plugins/opentelemetry/job/log/ConsoleNotes.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.jenkins.plugins.opentelemetry.job.log;
+
+import com.google.common.collect.ImmutableMap;
+import hudson.console.ConsoleNote;
+import io.jenkins.plugins.opentelemetry.semconv.JenkinsOtelSemanticAttributes;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utilities for extracting and reinserting {@link ConsoleNote}s.
+ * copied from https://github.com/jenkinsci/pipeline-cloudwatch-logs-plugin
```

Comment:
…and then modified from there. Candidate for a convenience API in `workflow-api`?

Reply:
That could be an idea. The helper in the Jenkins Otel Plugin changed since then but it can probably be useful for other. implementations of JEP 210